libs/components/core/src/lib/modules/sr-label/sr-label.directive.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DestroyRef,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit,
+  inject,
+} from '@angular/core';
+
+const SR_LABELS_CONTAINER_ID = 'sky-sr-labels-container';
+
+/**
+ * Adds the element to a screen reader only section of the body.
+ * This prevents components' DOM from including text only intended for screen readers.
+ */
+@Directive({
+  selector: '[skySrLabel]',
+  standalone: true,
+})
+export class SkySrLabelDirective implements OnInit {
+  public ngOnInit(): void {
+    this.#insertSrLabel();
+  }
+
+  @Input()
+  public set destroyRef(ref: DestroyRef) {
+    ref.onDestroy(() => this.#removeSrLabel());
+  }
+
+  @Input()
+  public set createLabel(value: boolean | undefined) {
+    this.#_createLabel = value ?? false;
+    this.#insertSrLabel();
+  }
+
+  public get createLabel(): boolean {
+    return this.#_createLabel;
+  }
+
+  readonly #elementRef = inject(ElementRef);
+  #_createLabel = false;
+
+  #insertSrLabel(): void {
```

Comment:
I would rename to `#updateScreenReaderLabel` (or `#updateLabel`) since it can both create and remove the label depending on a condition.